.github/workflows/release-please.yml

```diff
@@ -0,0 +1,123 @@
+on:
+  push:
+    branches:
+      - release/testnet
+      - release/integrationnet
+      - release/mainnet
+
+name: Release please
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+    name: Release please
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get commit message
+        id: get_commit_message
+        run: echo "COMMIT_MESSAGE=$(git log -1 --pretty=%B)" >> "$GITHUB_ENV"
+
+      - name: Check for chore(release)
+        id: check_commit_message
+        run: |
+          if [[ "$COMMIT_MESSAGE" == chore\(release\)* ]]; then
+            echo "skip=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "skip=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Setup Java and scala
+        if: steps.check_commit_message.outputs.skip == 'false'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          JAVA_VERSION: openjdk@1.11.0
+        uses: "./.github/templates/setup_java_and_scala"
+
+      - name: Test
+        if: steps.check_commit_message.outputs.skip == 'false'
+        run: sbt 'scalafixAll --check --rules OrganizeImports;scalafmtCheckAll;test'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get the release version
+        if: steps.check_commit_message.outputs.skip == 'false'
+        id: get_version
+        run: |
+          {
+            echo 'VERSION<<EOF'
+            npx commit-and-tag-version --dry-run | grep -oP 'tagging release v\K[0-9]+\.[0-9]+\.[0-9]+'
+            echo EOF
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Build Changelog
+        if: steps.check_commit_message.outputs.skip == 'false'
+        id: get_changelog
+        run: |
+          {
+            echo "CHANGELOG<<EOF"
+            npx commit-and-tag-version --dry-run | sed -n '/---/,/---/p' | sed '1d;$d'
+            echo EOF
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Set Git config
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+
+      - name: Bump version.sbt and create release tag
+        if: steps.check_commit_message.outputs.skip == 'false'
+        env:
+          VERSION: ${{ steps.get_version.outputs.VERSION }}
+        run: |
+          echo "Release-As: $VERSION" >> $GITHUB_STEP_SUMMARY
+
+          echo "ThisBuild / version := \"$VERSION\"" > version.sbt
```

Comment:
Eventually we could use `^([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?$ ` to just replace the version in file. It gives more flexibility but I am also fine with existing solution of replacing whole file.